### PR TITLE
refactor(plot): refactor callback scheduling

### PR DESCRIPTION
**Pull request by atsushi421**
## Description
This PR refactors `callback_sched` in Plot package.

## Related links
- JIRA ticket: https://tier4.atlassian.net/browse/T4PB-24805
- To reproduce: https://drive.google.com/drive/folders/1ir_2ixQzZkjzo_PP_PYj3BUO-exyRQko?usp=share_link

## Notes for reviewers
This PR organizes the commits of [this](https://github.com/tier4/CARET_analyze/pull/232).

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by hsgwa — CHANGES_REQUESTED** [review]
Here are some comments.
I think the behavior is good, but I had some concerns about the implementation.

File: src/caret_analyze/plot/visualize_lib/bokeh/color_selector.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        Get color.
+
+        Parameters
+        ----------
+        node_name : str, optional
+            node name, by default ''
+        cbg_name : str, optional
+            callback group name, by default ''
+        callback_name : str, optional
+            callback name, by default ''
```
[the remark this answers is not in the document]
Comment (atsushi421) [reply src/caret_analyze/plot/visualize_lib/bokeh/color_selector.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/240/commits/16a01e5cf1223c890b2725c199c149e15f3c726b.

File: src/caret_analyze/plot/visualize_lib/bokeh/color_selector.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            return ColorSelectorUnique()
+
+        if coloring_rule == 'callback':
+            return ColorSelectorCallback()
+
+        if coloring_rule == 'callback_group':
+            return ColorSelectorCbg()
+
+        if coloring_rule == 'node':
+            return ColorSelectorNode()
```
[the remark this answers is not in the document]
Comment (atsushi421) [reply src/caret_analyze/plot/visualize_lib/bokeh/color_selector.py:42]: Fixed in https://github.com/tier4/CARET_analyze/pull/240/commits/44c20e991c87e520f0c4219117ae3f7480a62a95.

File: src/caret_analyze/plot/visualize_lib/bokeh/color_selector.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+from bokeh.colors import Color, RGB
+
+import colorcet as cc
+
+
+class ColorSelectorFactory:
+    """Factory class to create an instance of ColorSelector."""
+
+    @staticmethod
+    def create_instance(coloring_rule: str):
```
[the remark this answers is not in the document]
Comment (atsushi421) [reply src/caret_analyze/plot/visualize_lib/bokeh/color_selector.py]: Addressed in https://github.com/tier4/CARET_analyze/commit/44c20e991c87e520f0c4219117ae3f7480a62a95.

File: src/caret_analyze/plot/callback_scheduling/callback_scheduling_plot.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        p = self.figure(xaxis_type, ywheel_zoom, full_legends, coloring_rule)
+        save(p, export_path, title=title, resources=CDN)
+
+    @staticmethod
+    def _get_callback_groups(target_objects: CallbackGroupTypes) -> List[CallbackGroup]:
+        callback_groups: List[CallbackGroup]
+
+        if isinstance(target_objects, (Application, Executor, Node)):
+            if target_objects.callback_groups is None:
+                raise InvalidArgumentError('target.callback_groups is None')
```
[the remark this answers is not in the document]
Comment (atsushi421) [reply src/caret_analyze/plot/callback_scheduling/callback_scheduling_plot.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/240/commits/472b091e771ea7eef2ed01f171f207ff7517d7e7.

File: src/caret_analyze/plot/callback_sched.py [hunk 5]
```diff
@@ -54,501 +37,13 @@ def callback_sched(
...
-    else:
-        frame_min = clipper.min_ns
-        frame_max = clipper.max_ns
-    apply_x_axis_offset(p, x_range_name, frame_min, frame_max)
-
-    rect_y = 0.0
-    rect_height = 0.3
-    rect_y_step = -1.5
-    callback_idx = 0
+    logger.warning("The 'callback_sched' interface is deprecated, "
```
[the remark this answers is not in the document]
Comment (atsushi421) [reply src/caret_analyze/plot/callback_sched.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/240/commits/f20dec5ea98449d0ca7a71417f1ed1ebeee44948.

**Review by hsgwa — APPROVED** [review]
LGTM!

**Pull request merged**